### PR TITLE
KEB: Combine CLS steps

### PR DESCRIPTION
**Pull request by skhalash**
**Description**

It's the final PR in a series of CLS steps, which combines them all together and enabled them in `main.go` (still hidden behind a feature flag)

Other changes proposed in this pull request:

- Fix FluentBit config indentation
- Pass the logger as a method argument (instead of a constructor argument) 
- Add all Azure regions in the region mapping routine
- Disable CLS steps for trial clusters 

**Related issue(s)**
See also #550  #536 #465 #565

### CODE REVIEW

File: components/kyma-environment-broker/cmd/broker/main.go [hunk 1]
```diff
@@ -339,8 +364,14 @@ func main() {
...
-			weight: 2,
-			step:   provisioning.NewLmsActivationStep(cfg.LMS, provisioning.NewProvideLmsTenantStep(lmsTenantManager, db.Operations(), cfg.LMS.Region, cfg.LMS.Mandatory)),
+			weight:   2,
+			step:     provisioning.NewClsProvisionStep(clsConfig, clsProvisioner, db.Operations()),
+			disabled: cfg.Cls.Disabled,
+		},
+		{
+			weight:   2,
+			step:     provisioning.NewLmsActivationStep(cfg.LMS, provisioning.NewProvideLmsTenantStep(lmsTenantManager, db.Operations(), cfg.LMS.Region, cfg.LMS.Mandatory)),
+			disabled: !cfg.Cls.Disabled,
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/cmd/broker/main.go:374]: How can I configure SKR to not use LMS and not use CLS ? there is such requirement

File: components/kyma-environment-broker/internal/process/upgrade_kyma/cls_binding.go [hunk 2]
```diff
@@ -131,7 +126,7 @@ func (s *ClsUpgradeBindStep) Run(operation internal.UpgradeKymaOperation, log lo
 		}
 	}
 
-	operation.InputCreator.SetLabel(kibanaURLLabelKey, overrideParams.KibanaUrl)
+	//operation.InputCreator.SetLabel(kibanaURLLabelKey, overrideParams.KibanaUrl)
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/process/upgrade_kyma/cls_binding.go:129]: do we need this kibana url?

File: components/kyma-environment-broker/cmd/broker/main.go [hunk 3]
```diff
@@ -386,6 +419,18 @@ func main() {
...
+		{
+			weight:   7,
+			step:     provisioning.NewClsBindStep(clsConfig, clsClient, db.Operations(), cfg.Database.SecretKey),
+			disabled: cfg.Cls.Disabled,
+		},
+
+		{
+			weight:   8,
+			step:     provisioning.NewClsAuditLogOverridesStep(db.Operations(), cfg.AuditLog, cfg.Database.SecretKey),
+			disabled: !cfg.Cls.Disabled,
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/cmd/broker/main.go:431]: shoule we disable CLS audit step if CLS is enabled

Comment (skhalash) [reply components/kyma-environment-broker/cmd/broker/main.go:431]: You are right, it should be the other way around. Will fix it

File: components/kyma-environment-broker/internal/process/upgrade_kyma/cls_binding.go (hunk 2 above)
Comment (skhalash) [reply components/kyma-environment-broker/internal/process/upgrade_kyma/cls_binding.go]: I checked with Benjamin and figured that they are not using those labels anymore. Will delete that field completely

File: components/kyma-environment-broker/cmd/broker/main.go (hunk 1 above)
Comment (skhalash) [reply components/kyma-environment-broker/cmd/broker/main.go:374]: Hmm, do we have such a requirement? I am not aware of it. What would you suggest in this case? Introducing a disabled flag for LMS?

File: components/kyma-environment-broker/internal/cls/deprovisioner.go [hunk 4]
```diff
@@ -38,7 +36,7 @@ type DeprovisionRequest struct {
 	Instance      servicemanager.InstanceKey
 }
 
-func (d *Deprovisioner) Deprovision(smClient servicemanager.Client, request *DeprovisionRequest) error {
+func (d *Deprovisioner) Deprovision(log logrus.FieldLogger, smClient servicemanager.Client, request *DeprovisionRequest) error {
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/cls/deprovisioner.go:39]: minor (could be fixed later): there is a convention in KEB that logger argument is the last one

File: components/kyma-environment-broker/cmd/broker/main.go (hunk 3 above)
Comment (skhalash) [reply components/kyma-environment-broker/cmd/broker/main.go:431]: Fixed in 0e36826e3993c43002bdbe7ec6dbf9a530607d1e

File: components/kyma-environment-broker/internal/process/upgrade_kyma/cls_binding.go (hunk 2 above)
Comment (skhalash) [reply components/kyma-environment-broker/internal/process/upgrade_kyma/cls_binding.go]: Fixed in 0e36826

File: components/kyma-environment-broker/cmd/broker/main.go (hunk 1 above)
Comment (piotrmiskiewicz) [reply components/kyma-environment-broker/cmd/broker/main.go:374]: sometimes LMS is not used (disabled), for example on stage and enabled only for given (configured) GA. Merging this PR would enable LMS for all stage SKRs which is not acceptable.

Comment (piotrmiskiewicz) [reply components/kyma-environment-broker/cmd/broker/main.go:374]: there is already a flag for LMS

Comment (piotrmiskiewicz) [reply components/kyma-environment-broker/cmd/broker/main.go:374]: the NewLmsActivationStep step uses it

Comment (piotrmiskiewicz) [reply components/kyma-environment-broker/cmd/broker/main.go:374]: oh, I missed your changes - it is ok as it is. the NewLmsActivationStep step does the work. no problem

File: components/kyma-environment-broker/cmd/broker/main.go (hunk 3 above)
Comment (rakesh-garimella) [reply components/kyma-environment-broker/cmd/broker/main.go]: fixed.

File: components/kyma-environment-broker/internal/cls/deprovisioner.go (hunk 4 above)
Comment (rakesh-garimella) [reply components/kyma-environment-broker/internal/cls/deprovisioner.go]: fixed

File: components/kyma-environment-broker/internal/cls/provisioner.go [hunk 5]
```diff
@@ -42,57 +40,54 @@ type ProvisionRequest struct {
...
 type ProvisionResult struct {
-	InstanceID            string
-	ProvisioningTriggered bool
-	Region                string
+	InstanceID string
+	Region     string
 }
 
-func (p *provisioner) Provision(smClient servicemanager.Client, request *ProvisionRequest) (*ProvisionResult, error) {
+func (p *provisioner) Provision(log logrus.FieldLogger, smClient servicemanager.Client, request *ProvisionRequest) (*ProvisionResult, error) {
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/cls/provisioner.go:47]: the same like for deprovision - log at the end

Comment (rakesh-garimella) [reply components/kyma-environment-broker/internal/cls/provisioner.go]: missed it. Fixed now.

**Review by piotrmiskiewicz — APPROVED** [review]
lgtm

**Pull request merged**